plugin/legend/legend.model.js

```diff
@@ -0,0 +1,7 @@
+import {Resource} from '../../core/resource';
```

Comment:
there is no need in plugin/legend, it doesnt contain any addition logic, please remove all this 3 files